src/field.rs

```diff
@@ -200,78 +200,86 @@ pub trait FieldElementWithInteger: FieldElement + From<Self::Integer> {
 
     /// Returns the integer representation of the multiplicative identity.
     fn one_integer() -> Self::Integer;
-}
 
-/// Methods common to all `FieldElementWithInteger` implementations that are private to the crate.
-pub(crate) trait FieldElementWithIntegerExt: FieldElementWithInteger {
-    /// Encode `input` as bitvector of elements of `Self`. Output is written into the `output` slice.
-    /// If `output.len()` is smaller than the number of bits required to respresent `input`,
-    /// an error is returned.
-    ///
-    /// # Arguments
+    /// Encode the integer `input` as a sequence of bits in two's complement representation, least
+    /// significant bit first, and then map each bit to a field element.
     ///
-    /// * `input` - The field element to encode
-    /// * `output` - The slice to write the encoded bits into. Least signicant bit comes first
-    fn fill_with_bitvector_representation(
-        input: &Self::Integer,
-        output: &mut [Self],
-    ) -> Result<(), FieldError> {
-        // Create a mutable copy of `input`. In each iteration of the following loop we take the
-        // least significant bit, and shift input to the right by one bit.
-        let mut i = *input;
-
-        let one = Self::one_integer();
-        for bit in output.iter_mut() {
-            let w = Self::from(i & one);
-            *bit = w;
-            i = i >> 1;
+    /// Returns an error if `input` cannot be represented with `bits` many bits, or if `bits`
+    /// is larger than the bit width of the field's modulus.
+    fn encode_into_bitvector_representation(
```

Comment:
I wonder if we could drop `_representation` here and in the decode method. Is `encode_into_bitvector` clear enough?

Reply:
I went with `encode_as_bitvector()` and `decode_bitvector()`.